### PR TITLE
GraphNG: Enable scale distribution configuration

### PR DESCRIPTION
**Pull request by dprokop**
Ref: https://github.com/grafana/grafana/issues/27848
This adds a custom editor of scale distribution and updates the UPlotScaleBuilder to use provided config.

The custom editor looks like this: 
Kapture 2020-12-08 at 14 18 28

Added placeholder for currently supported log32 and log1024 modes. These are currently not supported by uPlot([feature request](https://github.com/leeoniya/uPlot/issues/393)), as it only supports log2 and log10 scales.

@leeoniya if you could take a look at this that would be awesome. I haven't tracked down the issue, but apparently enabling log scales via Scale config described in https://github.com/leeoniya/uPlot/blob/master/dist/uPlot.d.ts#L393 does not work. See the comparison of angular vs graph ng.  GraphNG is rendered without y axis and no data is shown. This test is performed on the simplest CSV input for test data source:
image

Scale configuration is as bellow:

```
temp2.areaMI2
{time: undefined, distr: 3, log: 2, range: ƒ}
```

### CODE REVIEW

**Comment by leeoniya** [comment]
i'll take a look later today. but this demo works okay for 10 and 2: https://leeoniya.github.io/uPlot/demos/log-scales2.html

**Comment by dprokop** [comment]
> i'll take a look later today. but this demo works okay for 10 and 2: https://leeoniya.github.io/uPlot/demos/log-scales2.html

Yups, it does, that's where I took inspiration from :) I wonder if I'm missing anything.

**Review by ryantxu — APPROVED** [review]
:fire:

**Comment by leeoniya** [comment]
@dprokop this now works correctly for +Y values, but crashes and burns for 0 and negative values, because `Math.log()` is undefined for these values. i've added a few commits to uPlot that result in similar behavior to Flot, effectively excluding `<= 0` from scale ranging, and rendering their values an order of magnitude below the chart edge.

i plan to do a uPlot bump tomorrow or Thurs, so let's wait till that happens before landing this.

**Review by mckn — APPROVED** [review]
LGTM! I like the UI or this.

File: public/app/plugins/panel/graph3/ScaleDistributionEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          });
+        }}
+      />
+      {value.type === ScaleDistribution.Logarithmic && (
+        <Select
+          allowCustomValue={false}
+          autoFocus
+          options={LOG_DISTRIBUTION_OPTIONS}
+          value={value.log || 2}
+          prefix={'log'}
```
Comment (ryantxu) [inline public/app/plugins/panel/graph3/ScaleDistributionEditor.tsx:59]: Maybe prefix with base instead of log

**Review by ryantxu — APPROVED** [review]
Looks good

**Comment by ryantxu** [comment]
merged with master and changed "log" to "base" in the UI:
image

**Pull request merged**